browser-extension/mv2/src/background/background.js

```diff
@@ -1,3 +1,75 @@
+// import { startEventBatchWriter } from "./EventWriter";
+
+/* can be moved to a different file --- EventWriter.js */
+
+let eventsBatch = [];
+let batchWriterInterval = null;
+
+const LOCAL_ANALYTICS_KEY = "analytics_event";
+
+function addEventToBatch(event) {
```

Comment:
```suggestion
function addEvent(event) {
```
or anything relevant